### PR TITLE
MAGN-6138 Empty CBN cannot be re-position - Missed in code re factoring process

### PR DESCRIPTION
**Pull request by ramramps**
This feature was implemented in 0.7.5 and this was missed in the code refactoring process. 

This fix performs the below action :

Defers the deletion of empty code block
Deletes an empty CBN, only if Esc key is pressed inside that CBN
Enables creation of multiple CBN.
Records the deletion of an empty CBN. This helps the user to undo the deletion of that empty CBN.

- [x] @pboyer

### CODE REVIEW

**Comment by jnealb** [comment]
@ramramps @pboyer Actually this feature was not implemented in 7.5, rather it came in the early 7.6 builds before the refactoring. However it still qualifies as a regression, imo.

**Comment by Benglin** [comment]
Hang on, @ramramps, your original code changes were reverted with [my pull request](https://github.com/DynamoDS/Dynamo/pull/3521) intentionally (because pressing `Escape` to remove an empty CBN surprises some and that isn't what we want as a user experience). I don't think it we want that behavior, is there any issue that we see with moving empty CBN now? I can move an empty CBN that I've just created, I must be missing something obvious.

**Comment by ramramps** [comment]
@Benglin @lukechurch  - Ben, I tagged Luke here. It was Luke's idea earlier to have that feature. 
And yes, we cannot move empty code block node. That piece was removed as part of refactoring process,

**Comment by pboyer** [comment]
HI @Benglin & @ramramps & @lukechurch,

Please have a discussion ASAP and then I will review this.

File: src/DynamoCoreWpf/Views/CodeBlocks/CodeBlockEditor.xaml.cs [hunk 1]
```diff
@@ -489,14 +495,15 @@ private void DiscardChangesAndOptionallyRemoveNode(UndoRedoRecorder recorder)
                     recorder.PopFromUndoGroup(); // Pop off creation action.
 
                 // The empty code block node needs to be removed from workspace.
-                nodeViewModel.WorkspaceViewModel.Model.RemoveNode(nodeModel);
+                //Fix - MAGN - 6138 - Do not remove empty code block when clicked outside.               
+                //nodeViewModel.WorkspaceViewModel.Model.RemoveNode(nodeModel);
```
Comment (pboyer) [inline src/DynamoCoreWpf/Views/CodeBlocks/CodeBlockEditor.xaml.cs]: No need to hold comment this out.  Just delete it.

**Comment by pboyer** [comment]
One small fix, and then LGTM.

**Comment by ramramps** [comment]
@pboyer  - removed that comment. Please check and let me know.

**Comment by ramramps** [comment]
@pboyer  - Can I merge this request?

**Pull request merged**